alacritty/src/event.rs

```diff
@@ -411,8 +413,32 @@ impl<N: Notify + OnResize> Processor<N> {
                 GlutinEvent::RedrawEventsCleared => {
                     *control_flow = ControlFlow::Wait;
 
-                    if event_queue.is_empty() {
-                        return;
+                    let event_queue_empty = event_queue.is_empty();
+
+                    #[cfg(not(any(target_os = "macos", windows)))]
```

Comment:
It might just be because of the terrible formatting which can be fixed separately, but these are relatively big platform-dependent blocks. Might be worth extracting this into a separate method?

Reply:
It makes sense to extract it, I agree.

Reply:
This still isn't ideal. I think it would get much, much cleaner if the method was renamed to something more like `event_queue_empty` and was implemented on all platforms checking for `event_queue.is_empty()` and the wayland status.

Then we can just do `if self.event_queue_empty() { return }`.